librz/bin/bobj.c

```diff
@@ -517,6 +537,48 @@ RZ_IPI void rz_bin_object_filter_strings(RzBinObject *bo) {
 	}
 }
 
+/**
+ * \brief Put the given address on top of o's base address
+ */
 RZ_API ut64 rz_bin_object_addr_with_base(RzBinObject *o, ut64 addr) {
 	return o ? addr + o->baddr_shift : addr;
 }
+
+/* \brief Resolve the given address pair to a vaddr if possible
+ * returns vaddr, rebased with the baseaddr of bin, if va is enabled for bin,
+ * paddr otherwise
+ */
+RZ_API ut64 rz_bin_object_get_vaddr(RzBinObject *o, ut64 paddr, ut64 vaddr) {
+	rz_return_val_if_fail(o, UT64_MAX);
+
+	if (paddr == UT64_MAX) {
+		// everything we have is the vaddr
+		return vaddr;
+	}
+
+	/* hack to realign thumb symbols */
```

Comment:
I suggest to extract this Thumb hack into the separate function.

Reply:
To be honest, I would rather not. Since this is quite a dirty hack, the goal should be to eventually remove it from there completely (not in this pr since it just moves code). By putting it in a separate functions we would obfuscate a bit how dirty it is when looking at the `rz_bin_object_get_vaddr()` function.